scripts/scr_get_item_names/scr_get_item_names.gml

```diff
@@ -0,0 +1,806 @@
+#macro ITEM_NAME_NONE "(None)"
+#macro ITEM_NAME_ANY "(any)"
+
+/// @description This function returns an array with common empty or wildcard names as needed.
+/// @param {bool} _with_none - Whether to include ITEM_NAME_NONE in the list.
+/// @param {bool} _with_any - Whether to include ITEM_NAME_ANY in the list.
+function get_none_or_any_item_names(_with_none=false, _with_any=false) {
+    var _item_names = [];
+    if (_with_none) {
+        array_push(_item_names, ITEM_NAME_NONE);
+    }
+    if (_with_any) {
+        array_push(_item_names, ITEM_NAME_ANY);
+    }
+    return _item_names;
+}
+
+/// @description This function returns the hard-coded list of ranged weapons.
+/// @param {array} _item_names - The list of ranged weapons to append to.
+/// @returns {array} The appended list.
+function push_marine_ranged_weapons_item_names(_item_names) {
+    var item_count = 23
+    var initial_size = array_length(_item_names);
+    array_resize(_item_names, initial_size + item_count);
+
+    var index = initial_size;
+    _item_names[index++] = "Archeotech Laspistol";
+    _item_names[index++] = "Assault Cannon";
+    _item_names[index++] = "Bolt Pistol";
+    _item_names[index++] = "Bolter";
+    _item_names[index++] = "Stalker Pattern Bolter";
+    _item_names[index++] = "Combiflamer";
+    _item_names[index++] = "Flamer";
+    _item_names[index++] = "Heavy Bolter";
+    _item_names[index++] = "Heavy Flamer";
+    _item_names[index++] = "Hellrifle";
+    _item_names[index++] = "Incinerator";
+    _item_names[index++] = "Integrated Bolter";
+    _item_names[index++] = "Lascannon";
+    _item_names[index++] = "Lascutter";
+    _item_names[index++] = "Meltagun";
+    _item_names[index++] = "Missile Launcher";
+    _item_names[index++] = "Multi-Melta";
+    _item_names[index++] = "Autocannon";
+    _item_names[index++] = "Plasma Gun";
+    _item_names[index++] = "Plasma Pistol";
+    _item_names[index++] = "Sniper Rifle";
+    _item_names[index++] = "Storm Bolter";
+    _item_names[index++] = "Webber"; // 23
+    // future: not needed when we stop using GML's deprecated Copy-on-Write behavior
```

Comment:
I may be misunderstanding, but can't you use the `@` accessor? It enables the pointer behaviour case-per-case.

Reply:
Ah, I'm new to gml, and it's javascript-looking gets me to assume I know what I'm doing. I'm wasn't aware of the `@` accessor. I'll update to use it.

Reply:
I made this change in cf8aefae